### ISSUE TITLE
IllegalStateException: Start block missed in BlockIterator

### ISSUE THREAD

**Issue by ZakShearman** [Minestom/Minestom#1092]
This issue occurs if you call LivingEntity#getTargetBlockPosition for a player soon after they have logged in.

Full stack trace:
```
java.lang.IllegalStateException: Start block missed in BlockIterator
	at net.minestom.server.utils.block.BlockIterator.<init>(BlockIterator.java:166)
	at net.minestom.server.utils.block.BlockIterator.<init>(BlockIterator.java:231)
	at net.minestom.server.utils.block.BlockIterator.<init>(BlockIterator.java:272)
	at net.minestom.server.entity.LivingEntity.getTargetBlockPosition(LivingEntity.java:656)
	at com.github.empires.feature.HoveredBlockParticleFeature.spawnParticles(HoveredBlockParticleFeature.java:29)
	at net.minestom.server.timer.Task$Builder$1.get(Task.java:68)
	at net.minestom.server.timer.Task$Builder$1.get(Task.java:59)
	at net.minestom.server.timer.SchedulerImpl.handleTask(SchedulerImpl.java:92)
	at net.minestom.server.timer.SchedulerImpl.lambda$processTick$3(SchedulerImpl.java:56)
	at org.jctools.queues.MessagePassingQueueUtil.drain(MessagePassingQueueUtil.java:39)
	at org.jctools.queues.BaseMpscLinkedArrayQueue.drain(BaseMpscLinkedArrayQueue.java:638)
	at org.jctools.queues.MpscUnboundedArrayQueue.drain(MpscUnboundedArrayQueue.java:23)
	at org.jctools.queues.MpscUnboundedArrayQueue.drain(MpscUnboundedArrayQueue.java:63)
	at net.minestom.server.timer.SchedulerImpl.processTick(SchedulerImpl.java:53)
	at net.minestom.server.timer.SchedulerImpl.processTick(SchedulerImpl.java:39)
	at net.minestom.server.timer.SchedulerManager.processTick(SchedulerManager.java:19)
	at net.minestom.server.ServerProcessImpl$TickerImpl.tick(ServerProcessImpl.java:280)
	at net.minestom.server.thread.TickSchedulerThread.run(TickSchedulerThread.java:24)
```

Code:

```java
public class HoveredBlockParticleFeature implements Feature {

    @Override
    public void hook() {
        SchedulerManager schedulerManager = MinecraftServer.getSchedulerManager();

        schedulerManager.scheduleTask(this::spawnParticles, TaskSchedule.nextTick(), TaskSchedule.tick(4));
    }

    private void spawnParticles() {
        for (Player player : MinecraftServer.getConnectionManager().getOnlinePlayers()) {
            if (player.getInstance() == null) continue;

            Point targetPoint = player.getTargetBlockPosition(10);
            if (targetPoint == null) continue;
            targetPoint = targetPoint.add(0.5, 0.5, 0.5);

            for (double x = -0.5; x <= 0.5; x += 0.5) {
                for (double z = -0.5; z <= 0.5; z += 0.5) {
                    if (x == 0 && z == 0) continue; // center looks horrid
                    ParticlePacket packet = ParticleCreator.createParticlePacket(Particle.DUST, true,
                            targetPoint.x() + x, targetPoint.y() + 0.53, targetPoint.z() + z,
                            0, 0, 0, 0, 1,
                            binaryWriter -> {
                                binaryWriter.writeFloat(0);
                                binaryWriter.writeFloat(0);
                                binaryWriter.writeFloat(0);
                                binaryWriter.writeFloat(0.65f);
                            });

                    player.sendPacket(packet);
                }
            }

        }
    }
}
```

**Comment by BitesizedLion** [comment]
https://user-images.githubusercontent.com/42980888/169313692-ca866be0-a2d9-436a-b48a-368d736efc65.mp4

**Comment by doomslayer6000** [comment]
Same thing on `master-SNAPSHOT`
Just set player velocity and in some cases (idk reasons) it crashes in loop

![image](https://user-images.githubusercontent.com/111844000/190239231-5b18b353-5839-4c42-94c9-d88e138b5953.png)

**Comment by doomslayer6000** [comment]
> Same thing on `master-SNAPSHOT` Just set player velocity and in some cases (idk reasons) it crashes in loop
> 
> ![image](https://user-images.githubusercontent.com/111844000/190239231-5b18b353-5839-4c42-94c9-d88e138b5953.png)

I think it caused by big velocity

**Comment by Insprill** [comment]
I've noticed this happening if the entity's location is exactly 0,0 on the X and Z, so this might be related to that if it's being run before the position is set.

**Comment by iam4722202468** [comment]
```
            Vec v1 = new Vec(41.21612735985405, 21, -51.0);
            Vec v2 = new Vec(-0.7195944535812537, -0.5000002169949778, 0.6943945725414694);
            BlockIterator iterator = new BlockIterator(v1, v2, 0, 2);```

This causes the same issue

**Comment by iam4722202468** [comment]
Closed by #1509
